source/common/common/regex.h

```diff
@@ -40,13 +40,30 @@ class CompiledGoogleReMatcher : public CompiledMatcher {
     return result;
   }
 
-private:
+protected:
   const re2::RE2 regex_;
 };
 
+class CompiledGoogleRePartialMatcher : public CompiledGoogleReMatcher {
+public:
+  explicit CompiledGoogleRePartialMatcher(const std::string& regex, bool do_program_size_check)
+      : CompiledGoogleReMatcher(regex, do_program_size_check) {}
+
+  // CompiledGoogleReMatcher
+  bool match(absl::string_view value) const override {
+    return re2::RE2::PartialMatch(re2::StringPiece(value.data(), value.size()), regex_);
+  }
+};
+
 class GoogleReEngine : public Engine {
 public:
+  GoogleReEngine();
```

Comment:
Why is the default constructor needed?

Reply:
Oh, it is not needed. I just attempted to pass the CI and see if there if further issues in the patch. I will fix that.